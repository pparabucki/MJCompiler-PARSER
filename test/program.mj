program Program
	
	class A{
		int x[],y[];
	}
	
	const int pi = 3, e = 2;
		class B extends A{
		int i; int x[];
		int getValue(int a) int b; bool c;
		{ return this.i + this.x[0] + a; }
	}
	
	class C extends B{
		A theA;
		string a;
	}
{
	void main() A a; C c; int i; int x[];{
	a = new A;
	a.x = new int[5];
	a.y = new int[5];
	c = new C;
	c.theA = a;
	c.x = new int[5];
	x = new int[3];
	i=0;
	read(c.i);
	
	while(i<5) {
		read(c.x[i]); read(c.theA.x[i]); 
		i++;
	}
	print(c.getValue(c.theA.x[0]));
	}
	// dodatak
	b = TRUE;
	f = FAlSE;
	123DANAS;
	"OVO JE S
	TRING"
	'OVO J
	"
	'	
}