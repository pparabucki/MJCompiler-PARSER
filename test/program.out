INFO 1384860548110 #1 program
INFO 1384860548111 #15 Program
INFO 1384860548111 #7 class
INFO 1384860548111 #15 A
INFO 1384860548111 #41 {
INFO 1384860548111 #15 int
INFO 1384860548111 #15 x
INFO 1384860548111 #43 [
INFO 1384860548112 #44 ]
INFO 1384860548112 #37 ,
INFO 1384860548112 #15 y
INFO 1384860548112 #43 [
INFO 1384860548112 #44 ]
INFO 1384860548112 #36 ;
INFO 1384860548112 #42 }
INFO 1384860548113 #13 const
INFO 1384860548113 #15 int
INFO 1384860548113 #15 pi
INFO 1384860548113 #25 =
INFO 1384860548113 #16 3
INFO 1384860548113 #37 ,
INFO 1384860548114 #15 e
INFO 1384860548114 #25 =
INFO 1384860548114 #16 2
INFO 1384860548114 #36 ;
INFO 1384860548114 #7 class
INFO 1384860548115 #15 B
INFO 1384860548115 #8 extends
INFO 1384860548115 #15 A
INFO 1384860548115 #41 {
INFO 1384860548115 #15 int
INFO 1384860548115 #15 i
INFO 1384860548115 #36 ;
INFO 1384860548115 #15 int
INFO 1384860548116 #15 x
INFO 1384860548116 #43 [
INFO 1384860548116 #44 ]
INFO 1384860548116 #36 ;
INFO 1384860548116 #15 int
INFO 1384860548116 #15 getValue
INFO 1384860548116 #39 (
INFO 1384860548117 #15 int
INFO 1384860548117 #15 a
INFO 1384860548117 #40 )
INFO 1384860548117 #15 int
INFO 1384860548117 #15 b
INFO 1384860548117 #36 ;
INFO 1384860548117 #15 bool
INFO 1384860548118 #15 c
INFO 1384860548118 #36 ;
INFO 1384860548118 #41 {
INFO 1384860548118 #14 return
INFO 1384860548118 #15 this
INFO 1384860548118 #38 .
INFO 1384860548118 #15 i
INFO 1384860548118 #20 +
INFO 1384860548119 #15 this
INFO 1384860548119 #38 .
INFO 1384860548119 #15 x
INFO 1384860548119 #43 [
INFO 1384860548120 #16 0
INFO 1384860548120 #44 ]
INFO 1384860548120 #20 +
INFO 1384860548120 #15 a
INFO 1384860548120 #36 ;
INFO 1384860548120 #42 }
INFO 1384860548120 #42 }
INFO 1384860548121 #7 class
INFO 1384860548121 #15 C
INFO 1384860548121 #8 extends
INFO 1384860548121 #15 B
INFO 1384860548121 #41 {
INFO 1384860548121 #15 A
INFO 1384860548121 #15 theA
INFO 1384860548122 #36 ;
INFO 1384860548122 #15 string
INFO 1384860548122 #15 a
INFO 1384860548122 #36 ;
INFO 1384860548122 #42 }
INFO 1384860548122 #41 {
INFO 1384860548122 #12 void
INFO 1384860548122 #15 main
INFO 1384860548122 #39 (
INFO 1384860548123 #40 )
INFO 1384860548123 #15 A
INFO 1384860548123 #15 a
INFO 1384860548123 #36 ;
INFO 1384860548123 #15 C
INFO 1384860548123 #15 c
INFO 1384860548123 #36 ;
INFO 1384860548123 #15 int
INFO 1384860548123 #15 i
INFO 1384860548124 #36 ;
INFO 1384860548124 #15 int
INFO 1384860548124 #15 x
INFO 1384860548124 #43 [
INFO 1384860548124 #44 ]
INFO 1384860548124 #36 ;
INFO 1384860548124 #41 {
INFO 1384860548124 #15 a
INFO 1384860548125 #25 =
INFO 1384860548125 #9 new
INFO 1384860548125 #15 A
INFO 1384860548125 #36 ;
INFO 1384860548125 #15 a
INFO 1384860548125 #38 .
INFO 1384860548125 #15 x
INFO 1384860548125 #25 =
INFO 1384860548126 #9 new
INFO 1384860548126 #15 int
INFO 1384860548126 #43 [
INFO 1384860548126 #16 5
INFO 1384860548126 #44 ]
INFO 1384860548126 #36 ;
INFO 1384860548126 #15 a
INFO 1384860548126 #38 .
INFO 1384860548126 #15 y
INFO 1384860548127 #25 =
INFO 1384860548127 #9 new
INFO 1384860548127 #15 int
INFO 1384860548127 #43 [
INFO 1384860548127 #16 5
INFO 1384860548127 #44 ]
INFO 1384860548127 #36 ;
INFO 1384860548128 #15 c
INFO 1384860548128 #25 =
INFO 1384860548128 #9 new
INFO 1384860548128 #15 C
INFO 1384860548128 #36 ;
INFO 1384860548128 #15 c
INFO 1384860548128 #38 .
INFO 1384860548128 #15 theA
INFO 1384860548129 #25 =
INFO 1384860548129 #15 a
INFO 1384860548129 #36 ;
INFO 1384860548129 #15 c
INFO 1384860548129 #38 .
INFO 1384860548129 #15 x
INFO 1384860548129 #25 =
INFO 1384860548129 #9 new
INFO 1384860548130 #15 int
INFO 1384860548130 #43 [
INFO 1384860548130 #16 5
INFO 1384860548130 #44 ]
INFO 1384860548130 #36 ;
INFO 1384860548130 #15 x
INFO 1384860548130 #25 =
INFO 1384860548130 #9 new
INFO 1384860548130 #15 int
INFO 1384860548131 #43 [
INFO 1384860548131 #16 3
INFO 1384860548131 #44 ]
INFO 1384860548131 #36 ;
INFO 1384860548131 #15 i
INFO 1384860548131 #25 =
INFO 1384860548131 #16 0
INFO 1384860548131 #36 ;
INFO 1384860548132 #3 read
INFO 1384860548132 #39 (
INFO 1384860548132 #15 c
INFO 1384860548132 #38 .
INFO 1384860548132 #15 i
INFO 1384860548132 #40 )
INFO 1384860548132 #36 ;
INFO 1384860548132 #10 while
INFO 1384860548132 #39 (
INFO 1384860548132 #15 i
INFO 1384860548133 #30 <
INFO 1384860548133 #16 5
INFO 1384860548133 #40 )
INFO 1384860548133 #41 {
INFO 1384860548133 #3 read
INFO 1384860548133 #39 (
INFO 1384860548133 #15 c
INFO 1384860548134 #38 .
INFO 1384860548134 #15 x
INFO 1384860548134 #43 [
INFO 1384860548134 #15 i
INFO 1384860548134 #44 ]
INFO 1384860548134 #40 )
INFO 1384860548134 #36 ;
INFO 1384860548135 #3 read
INFO 1384860548135 #39 (
INFO 1384860548135 #15 c
INFO 1384860548135 #38 .
INFO 1384860548135 #15 theA
INFO 1384860548135 #38 .
INFO 1384860548135 #15 x
INFO 1384860548135 #43 [
INFO 1384860548135 #15 i
INFO 1384860548135 #44 ]
INFO 1384860548136 #40 )
INFO 1384860548136 #36 ;
INFO 1384860548136 #15 i
INFO 1384860548136 #34 ++
INFO 1384860548136 #36 ;
INFO 1384860548136 #42 }
INFO 1384860548136 #2 print
INFO 1384860548136 #39 (
INFO 1384860548136 #15 c
INFO 1384860548137 #38 .
INFO 1384860548137 #15 getValue
INFO 1384860548137 #39 (
INFO 1384860548137 #15 c
INFO 1384860548137 #38 .
INFO 1384860548137 #15 theA
INFO 1384860548138 #38 .
INFO 1384860548138 #15 x
INFO 1384860548138 #43 [
INFO 1384860548138 #16 0
INFO 1384860548138 #44 ]
INFO 1384860548138 #40 )
INFO 1384860548138 #40 )
INFO 1384860548138 #36 ;
INFO 1384860548138 #42 }
INFO 1384860548139 #15 b
INFO 1384860548139 #25 =
INFO 1384860548139 #18 TRUE
INFO 1384860548139 #36 ;
INFO 1384860548139 #15 f
INFO 1384860548139 #25 =
INFO 1384860548139 #15 FAlSE
INFO 1384860548139 #36 ;
INFO 1384860548140 #36 ;
INFO 1384860548140 #19 "OVO JE S
	TRING"
INFO 1384860548140 #17 'OVO J'E'
INFO 1384860548141 #42 }
